libafl_libfuzzer/libafl_libfuzzer_runtime/src/fuzz.rs

```diff
@@ -135,33 +168,80 @@ fn fuzz_single(
+    let i2s = SkippableStage::new(i2s, |_| (!mutator_status.custom_mutation).into());
+    let cm_i2s = StdMutationalStage::new(unsafe {
+        LLVMCustomMutator::mutate_unchecked(StdScheduledMutator::new(tuple_list!(
+            I2SRandReplace::new()
+        )))
+    });
+    let cm_i2s = SkippableStage::new(cm_i2s, |_| mutator_status.custom_mutation.into());
 
     // Setup a MOPT mutator
-    let mutator = StdMOptMutator::new(
+    // TODO configure with mutation stacking options from libfuzzer, use tokens from dictionary
+    let std_mutator = StdMOptMutator::new(
         &mut state,
         havoc_mutations(), // TODO .merge(tokens_mutations()),
         7,
         5,
     )?;
 
-    // TODO support
-    // let grimoire_mutator = StdScheduledMutator::with_max_stack_pow(
-    //     tuple_list!(
-    //         GrimoireExtensionMutator::new(),
-    //         GrimoireRecursiveReplacementMutator::new(),
-    //         GrimoireStringReplacementMutator::new(),
-    //         // give more probability to avoid large inputs
-    //         GrimoireRandomDeleteMutator::new(),
-    //         GrimoireRandomDeleteMutator::new(),
-    //     ),
-    //     3,
-    // );
-    // let grimoire = StdMutationalStage::new(grimoire_mutator);
-    // let skippable_grimoire = SkippableStage::new(grimoire, |_s| opt.grimoire.into());
-
-    let power = StdPowerMutationalStage::new(mutator, &edges_observer);
+    let std_power = StdPowerMutationalStage::new(std_mutator, &edges_observer);
+    let std_power = SkippableStage::new(std_power, |_| mutator_status.std_mutational.into());
+
+    // for custom mutator and crossover, each have access to the LLVMFuzzerMutate -- but it appears
+    // that this method doesn't normally offer stacked mutations where one may expect them
+    // we offer stacked mutations since this appears to be expected; see:
+    // https://github.com/google/fuzzing/blob/bb05211c12328cb16327bb0d58c0c67a9a44576f/docs/structure-aware-fuzzing.md#example-compression
+    // additionally, we perform mutation and crossover in two separate stages due to possible
+    // errors introduced by incorrectly handling custom mutations; see explanation below
+
+    // a custom mutator is defined
+    // note: in libfuzzer, crossover is enabled by default, but this appears to be unintended
+    // and erroneous if custom mutators are defined as it inserts bytes from other test cases
+    // without performing the custom mutator's preprocessing beforehand
+    // we opt not to use crossover in the LLVMFuzzerMutate and instead have a second crossover pass,
+    // though it is likely an error for fuzzers to provide custom mutators but not custom crossovers
+    let custom_mutator = unsafe {
+        LLVMCustomMutator::mutate_unchecked(StdMOptMutator::new(
+            &mut state,
+            havoc_mutations_no_crossover(), // TODO .merge(tokens_mutations()),
```

Comment:
using tokens_mutations should be straight forward as well, you don't need to implement a new token parsing - just use the one from LibAFL directly.
It's already pretty optimized for fuzzbench etc